### PR TITLE
[GEOT-5790] dbtype and filetype should be marked as 'program'-level parameters

### PR DESCRIPTION
**Pull request by tbarsballe**
See https://osgeo-org.atlassian.net/browse/GEOT-5790

Adds LEVEL="program" to DBTYPE / DSTYPE / FILETYPE parameters for all the various datastore implementations.

Ultimately intended to fix https://osgeo-org.atlassian.net/browse/GEOS-8010

### CODE REVIEW

File: modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/ShapefileDataStoreFactory.java [hunk 1]
```diff
@@ -82,7 +82,7 @@
      */
     public static final Param FILE_TYPE = new Param("filetype", String.class,
             "Discriminator for directory stores", false, "shapefile", new KVP(Param.LEVEL,
-                    "advanced"));
+                    "program"));
```
Comment (tbarsballe) [inline modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/ShapefileDataStoreFactory.java]: This was advanced before, should it really be changed to program?

File: modules/library/jdbc/src/main/java/org/geotools/data/jdbc/datasource/DBCPDataSourceFactory.java [hunk 2]
```diff
@@ -39,8 +41,8 @@
  */
 public class DBCPDataSourceFactory extends AbstractDataSourceFactorySpi {
 
-    public static final Param DSTYPE = new Param("dstype", String.class,
-            "Must be DBCP", false);
+    public static final Param DSTYPE = new Param("dstype", String.class, "Must be DBCP", false,
+            null, Collections.singletonMap(Parameter.LEVEL, "program"));
```
Comment (tbarsballe) [inline modules/library/jdbc/src/main/java/org/geotools/data/jdbc/datasource/DBCPDataSourceFactory.java]: Not "DBTYPE", but seems to have the same purpose; presuming we should add this one too.

File: modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/ShapefileDataStoreFactory.java (hunk 1 above)
Comment (jodygarnett) [inline modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/ShapefileDataStoreFactory.java]: yes, sorry I added program later when it became obvious some things were really only machine editable

**Comment by bradh** [comment]
Would those labels ("advanced", "program", ....) be better as constants?

**Pull request merged**